crates/nu-command/src/filters/where_.rs

```diff
@@ -35,7 +35,10 @@ not supported."#
             ])
             .required(
                 "row_condition",
-                SyntaxShape::RowCondition,
+                SyntaxShape::OneOf(vec![
+                    SyntaxShape::RowCondition,
+                    SyntaxShape::Closure(Some(vec![SyntaxShape::Any])),
+                ]),
```

Comment:
Regarding #16129 ,

Do we need the `Closure` here if we already manually override the expression type to make it compatible?